o-fish-ios/Views/ReportFlow/ChooseScreens/ChooseBusiness/ChooseBusinessView.swift

```diff
@@ -0,0 +1,208 @@
+        static let padding: CGFloat = 16.0
+    }
+
+    /// Interface
+
+    var body: some View {
+        VStack(spacing: Dimensions.padding / 2) {
+
+            SearchBarView(searchText: searchTextBinding, placeholder: "Choose Business")
+
+            ScrollView {
+                VStack(spacing: .zero) {
+
+                    VStack(alignment: .leading, spacing: .zero) {
+                        SectionButton(title: "New Business", systemImageName: "plus", action: newBusinessClicked)
+                            .padding(.vertical, Dimensions.padding)
+
+                        Divider()
+                    }
+
+                    stateView()
+                }
+            }
+                .padding(.horizontal, Dimensions.padding)
+        }
+
+            .onAppear(perform: onAppear)
+            .onDisappear(perform: onDisappear)
+    }
+
+    var searchTextBinding: Binding<String> {
+        Binding<String>(
+            get: { self.searchText },
+            set: {
+                self.searchText = $0
+
+                self.searchDebouncer.invalidate()
+                self.searchDebouncer.handler = { self.loadFilteredData(searchText: self.searchText) }
+                self.searchDebouncer.call()
+            }
+        )
+    }
+
+    private func stateView() -> some View {
+        switch state {
+        case . loading:
+            return AnyView(ActivityIndicator(isAnimating: .constant(true), style: .medium)
+                .padding(.top, Dimensions.padding * 2))
+
+        case .empty:
+            return AnyView(EmptyStateView(searchWord: searchText))
+
+        case .loaded(let items):
+            return AnyView(listView(items: items))
+        }
+    }
+
+    func listView(items: [BusinessPickerData]) -> some View {
+        VStack(spacing: .zero) {
+            ForEach(items) { item in
+                Button(action: { self.searchItemClicked(item: item) }) {
+                    BusinessPickerDataView(item: item)
+                }
+            }
+        }
+    }
+
+    /// Actions
+
+    private func onAppear() {
+        loadFilteredData(searchText: "")
+    }
+
+    private func onDisappear() {
+        notificationTokenSearch?.invalidate()
+    }
+
+    private func newBusinessClicked() {
+        self.selectedItem = .notSelected
+        self.isAutofillItem = false
+        dismiss()
+    }
+
+    private func searchItemClicked(item: BusinessPickerData) {
+        selectedItem = item
+        dismiss()
+    }
+
+    private func dismiss() {
+        presentationMode.wrappedValue.dismiss()
+    }
+
+    /// Logic
+
+    private func loadFilteredData(searchText: String) {
+        let realmFilteredReports = RealmConnection.realm?.objects(Report.self)
+        var result = realmFilteredReports
+
+        if searchText != "" {
+            let predicate = NSPredicate(format: "vessel.lastDelivery.business CONTAINS[cd] %@ OR vessel.lastDelivery.location CONTAINS[cd] %@",
+                searchText, searchText)
```

Comment:
Is it really needed, using the same variable "searchText" twice?

Reply:
yes, it is using two %@ as arguments, 
if keep only one that last one will be (null)